src/wallet/wallet.cpp

```diff
@@ -2841,11 +2836,9 @@ int CMerkleTx::GetDepthInMainChainINTERNAL(const CBlockIndex* &pindexRet) const
         return 0;
 
     // Make sure the merkle branch connects to this block
-    if (!fMerkleVerified)
+    if (nIndex == -1)
```

Comment:
Why is this added?  Its already done above no?
Or is `nIndex` somehow modified in another thread?